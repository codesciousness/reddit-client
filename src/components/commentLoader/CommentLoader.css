.CommentLoader {
    background: rgba(255, 255, 255, 0.7);
    border-radius: 20px;
    box-shadow: 0 5px 10px rgba(145, 145, 165, 0.6);
    margin: 2rem auto;
    padding: 1rem 2rem;
}

.CommentLoader > * {
    margin: 0.5rem 0;
}

.CommentLoader__author strong {
    color:lightgray;
    background: lightgray;
    border-radius: 5px;
    opacity: 0.7;
}

.CommentLoader__body {
    color:lightgray;
    background: lightgray;
    border-radius: 5px;
    opacity: 0.7;
    overflow: hidden;
}

.CommentLoader__ups strong {
    color:lightgray;
    background: lightgray;
    border-radius: 5px;
    opacity: 0.7;
}