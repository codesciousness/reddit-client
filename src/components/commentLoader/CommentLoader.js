import React from 'react';
import './CommentLoader.css';

const CommentLoader = () => {
    return (
        <section className="CommentLoader">
            <p className="CommentLoader__author"><strong>Comment author</strong> | <strong>Post time</strong></p>
            <p className="CommentLoader__body">Comment body</p>
            <p className="CommentLoader__ups"><i className="fas fa-arrow-up"></i> <strong>Ups#</strong> Ups</p>
        </section>
    );
}

export default CommentLoader;