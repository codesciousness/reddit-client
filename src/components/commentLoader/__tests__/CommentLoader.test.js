import React from 'react';
import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import CommentLoader from '../CommentLoader';

describe('<CommentLoader />', () => {
    test('CommentLoader component is defined', () => {
        expect(<CommentLoader />).toBeDefined();
    });
    test('matches the snapshot', () => {
        const component = create(<CommentLoader />);
        expect(component.toJSON()).toMatchSnapshot();
    });
    test('renders correctly', () => {
        const { container } = render(<CommentLoader />);
        expect(container.querySelector('.CommentLoader')).toBeInTheDocument();
        expect(screen.getByText('Comment author')).toBeInTheDocument();
        expect(screen.getByText('Post time')).toBeInTheDocument();
        expect(screen.getByText('Comment body')).toBeInTheDocument();
        expect(screen.getByText('Ups#')).toBeInTheDocument();
    });
});