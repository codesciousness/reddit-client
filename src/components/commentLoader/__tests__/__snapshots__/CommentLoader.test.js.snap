// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentLoader /> matches the snapshot 1`] = `
<section
  className="CommentLoader"
>
  <p
    className="CommentLoader__author"
  >
    <strong>
      Comment author
    </strong>
     | 
    <strong>
      Post time
    </strong>
  </p>
  <p
    className="CommentLoader__body"
  >
    Comment body
  </p>
  <p
    className="CommentLoader__ups"
  >
    <i
      className="fas fa-arrow-up"
    />
     
    <strong>
      Ups#
    </strong>
     Ups
  </p>
</section>
`;
